### ISSUE TITLE
replace neural mesh renderer with pytorch3d

### ISSUE THREAD

**Issue by YunjieYu** [elliottwu/unsup3d#38]
Hi, wu!

I‘m sorry to disturb you again. Inspired from your repo, I just replace the neural renderer with pytorch3d. Specially, I choose MeshRenderer in pytorch3d.

I first eliminated the inconsistency between NMR and PyTorch3D coordinate systems by overriding **_PerspectiveCameras_** class in pytorch3d. However, I found that the noise for shape is so big that the network can not learn a reasonable face shape. Note that I have trained the model with 100 epochs. The figure below is a comparison between NMR and PyTorch3D.

NMR:
![1](https://user-images.githubusercontent.com/44226851/141252347-6834310b-6f44-4976-b34a-8f6bb3e02936.png)
                                                                                     
PyTorch3D:
![2](https://user-images.githubusercontent.com/44226851/141252374-ab54e8a5-284a-46c6-b477-7368a0314dcd.png)

RasterizationSettings is showed as below:
```
     # hard rasterization
     raster_settings = RasterizationSettings(
            image_size=self.image_size,
            blur_radius=0.0,
            faces_per_pixel=1,
            bin_size=0
        )
```

I noticed that you mentioned one can add a smoothing loss to the depth map for alleviating the noisy depth problem in #9. So can you provide the specific form of such a smoothing loss? I really don't have a clue about this. 

Thank you very much!

**Comment by YunjieYu** [comment]
I close this issue because the problem has been solved.

**Comment by abhishekgahlot** [comment]
@YunjieYu Would you be able to share the code or branch where you successfully migrated to pytorch3d. I want to use unsup3d as well but on CPU only machine with a rendering animation.

**Comment by nihaomiao** [comment]
@YunjieYu Would you like to share your codes here? Thanks a lot!
